tapdb/universe_perf_test.go

```diff
@@ -0,0 +1,485 @@
+package tapdb
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/lightninglabs/taproot-assets/tapdb/sqlc"
+	"github.com/lightningnetwork/lnd/clock"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	numAssets         = 100
+	numLeavesPerTree  = 300
+	numEventsPerAsset = 100
+	numQueries        = 100
+)
+
+// longTestScale is the scale factor for long tests.
+const longTestScale = 5
+
+// Common indices for universe tables.
+const (
+	// createUniverseRootIndex optimizes lookups on universe roots.
+	createUniverseRootIndex = `
+		CREATE INDEX IF NOT EXISTS idx_universe_roots_composite
+		ON universe_roots(
+			namespace_root,
+			proof_type,
+			asset_id
+		);`
+
+	// createUniverseLeavesIndex optimizes universe leaf queries.
+	createUniverseLeavesIndex = `
+		CREATE INDEX IF NOT EXISTS idx_universe_leaves_composite
+		ON universe_leaves(
+			leaf_node_namespace,
+			universe_root_id,
+			leaf_node_key
+		);`
+
+	// createMSMTNodesIndex optimizes MSSMT node lookups.
+	createMSMTNodesIndex = `
+		CREATE INDEX IF NOT EXISTS idx_mssmt_nodes_composite
+		ON mssmt_nodes(
+			namespace,
+			key,
+			hash_key,
+			sum
+		);`
+)
+
+// executeSQLStatements executes a list of SQL statements with logging.
+func executeSQLStatements(
+	t *testing.T,
```

Comment:
See this file for our preferred formatting style (which implicitly we try to observe in all our Golang projects): https://github.com/lightningnetwork/lnd/blob/master/docs/code_formatting_rules.md#wrapping-long-function-definitions